core/src/blocktree.rs

```diff
@@ -357,6 +380,130 @@ impl Blocktree {
         Ok(())
     }
 
+    #[cfg(feature = "erasure")]
+    pub fn insert_data_blobs<I>(&self, new_blobs: I) -> Result<()>
+    where
+        I: IntoIterator,
+        I::Item: Borrow<Blob>,
+    {
+        let mut write_batch = self.db.batch()?;
+        // A map from slot to a 2-tuple of metadata: (working copy, backup copy),
+        // so we can detect changes to the slot metadata later
+        let mut slot_meta_working_set = HashMap::new();
+        let mut erasure_meta_working_set = HashMap::new();
+        let new_blobs: Vec<_> = new_blobs.into_iter().collect();
+        let mut prev_inserted_blob_datas = HashMap::new();
+
+        for blob in new_blobs.iter() {
+            let blob = blob.borrow();
+            let blob_slot = blob.slot();
+            let parent_slot = blob.parent();
+
+            // Check if we've already inserted the slot metadata for this blob's slot
+            let entry = slot_meta_working_set.entry(blob_slot).or_insert_with(|| {
+                // Store a 2-tuple of the metadata (working copy, backup copy)
+                if let Some(mut meta) = self
+                    .meta(blob_slot)
+                    .expect("Expect database get to succeed")
+                {
+                    // If parent_slot == std::u64::MAX, then this is one of the dummy metadatas inserted
+                    // during the chaining process, see the function find_slot_meta_in_cached_state()
+                    // for details
+                    if meta.parent_slot == std::u64::MAX {
+                        meta.parent_slot = parent_slot;
+                        // Set backup as None so that all the logic for inserting new slots
+                        // still runs, as this placeholder slot is essentially equivalent to
+                        // inserting a new slot
+                        (Rc::new(RefCell::new(meta.clone())), None)
+                    } else {
+                        (Rc::new(RefCell::new(meta.clone())), Some(meta))
+                    }
+                } else {
+                    (
+                        Rc::new(RefCell::new(SlotMeta::new(blob_slot, parent_slot))),
+                        None,
+                    )
+                }
+            });
+
+            let slot_meta = &mut entry.0.borrow_mut();
+
+            // This slot is full, skip the bogus blob
+            if slot_meta.is_full() {
+                continue;
+            }
+
+            // handle erasure meta info
+            let erasure_meta_entry =
```

Comment:
Can we wrap this block in a cfg flag instead of duplicating the entire function?

Reply:
Yes, i'm trying to extract the erasure related code into methods that will be no-ops then the erasure feature is turned off.

Reply:
this didn't work out by the way I'm just wrapping the blocks.